src/file_pipe_log/pipe.rs

```diff
@@ -382,7 +391,10 @@ impl<F: FileSystem> SinglePipe<F> {
             (files.len(), tail)
         };
         let purged_len = purged_files.len();
-        if purged_len > 0 {
+        let recycled_len = if purged_len == 0 {
+            // No need to purge files, return the count of recycled files directly.
+            self.recycled_files.read().len()
```

Comment:
No need to update too?

Reply:
IMO, the `rotate` will also cost recycled files if there exists some. And if we do not update the metrics on `recycled_files.len()` by the period `purge` progress, the count of `recycled_files.len()` won't be fresh as expected.

So, I delay the update of `recycled_files.len()` in `rotate` to `purge` at here.

Reply:
How about update it in rotate? You can break new_file into recycle_file -> Option<File> and new_file -> Result<File>.